### PR TITLE
Add WithMetricNameTransform as option for NewRecursiveMetricsFromTags

### PR DESCRIPTION
**Pull request by joreetz-otto**

### CODE REVIEW

File: v0/pkg/prometheus/gen/collector.go [hunk 1]
```diff
@@ -40,39 +54,58 @@ func NewRecursiveMetricsFromTags(tagged interface{}, opts ...RecursiveMetricsOpt
...
 	default:
-		transform = func(value string) (labelName string) {
-			for _, trans := range transforms {
+		metricNameTransform = func(value string) (metricName string) {
+			for _, trans := range metricNameTransforms {
 				value = trans(value)
 			}
 			return value
 		}
 	}
```
Comment (joreetz-otto) [inline v0/pkg/prometheus/gen/collector.go:97]: I was not able to extract this as a function :'(

File: internal/collector/collector.go [hunk 2]
```diff
@@ -40,7 +36,10 @@ func makeGenerated(i int, tag string, f reflect.StructField, parent string, labe
 	switch prom[0] {
 	case "CounterVec":
 		// FIXME: This could result in overlapping prefixes
-		namePrefix := metricNameCleaner.Replace(parent)
+		fmt.Println(parent)
```
Comment (abergmeier) [inline internal/collector/collector.go]: Probably need to remove all `Println`s

File: v0/pkg/prometheus/gen/collector.go [hunk 3]
```diff
@@ -25,10 +25,24 @@ func WithLabelNameTransform(fun types.LabelNameTransformer) RecursiveMetricsOpti
...
 }
 
+// WithLabelNameTransform creates an Option for transforming Prometheus label names
```
Comment (abergmeier) [inline v0/pkg/prometheus/gen/collector.go]: Comment is wrong.

File: v0/pkg/prometheus/gen/collector_test.go [hunk 4]
```diff
@@ -117,8 +119,13 @@ func TestUpdateSimpleTransformation(t *testing.T) {
 }
 
 func TestUpdateFull(t *testing.T) {
-	col, upd := NewRecursiveMetricsFromTags(&full)
+	col, upd := NewRecursiveMetricsFromTags(&full, WithMetricNameTransform(
```
Comment (abergmeier) [inline v0/pkg/prometheus/gen/collector_test.go:130]: I think it would be better to add a new Test, so we test the metric name transform separately.

Comment (joreetz-otto) [reply v0/pkg/prometheus/gen/collector_test.go:130]: @abergmeier 
This wont work as we need some basic metric name transformation. I removed the metricNameCleaner in internal/collector/collector.go so the default test cant work.

Comment (joreetz-otto) [reply v0/pkg/prometheus/gen/collector_test.go:130]: I had to add it to even more tests to get them working

**Review by AndreasBergmeier6176 — APPROVED** [review]

**Pull request merged**